tests/job_applications/tests.py

```diff
@@ -409,10 +408,14 @@ def test_with_jobseeker_eligibility_diagnosis(self):
         qs = JobApplication.objects.with_jobseeker_eligibility_diagnosis().get(pk=job_app.pk)
         assert qs.jobseeker_eligibility_diagnosis == diagnosis.pk
 
+    # @pytest.mark.skip(reason="This should not happen anymore. FIXME.")
     def test_with_eligibility_diagnosis_criterion(self):
-        job_app = JobApplicationFactory(with_approval=True, eligibility_diagnosis=None)
+        job_app = JobApplicationFactory(with_approval=True)
         diagnosis = EligibilityDiagnosisFactory(job_seeker=job_app.job_seeker, created_at=timezone.now())
 
+        job_app.eligibility_diagnosis = diagnosis
+        job_app.save()
```

Comment:
Créer les objets dans le bon ordre évite une requête SQL.
```suggestion
        diagnosis = EligibilityDiagnosisFactory(created_at=timezone.now())
        job_app = JobApplicationFactory(
            eligibility_diagnosis=diagnosis,
            job_seeker=diagnosis.job_seeker,
            with_approval=True,
        )

        job_app.eligibility_diagnosis = diagnosis
        job_app.save()
```